### DISCUSSION TITLE
Passing variables to php mailer script

### DISCUSSION THREAD

**Discussion by charco66** [ruffle-rs/ruffle#4337] [Q&A] [answered]
Hi Ruffle developers,

I have many educational flash quiz programs that pass the quiz results as variables to a php mailer that sends them to the teacher (me). However, this does not work in Ruffle. The php script does not receive/recognise the information. Is there a work around I can use?

The variables are loaded into a moviclip called "summary".
*******************************************************************
function fillForm() {
	// displays the summary results
	summary.myName.text = me;
	summary.score.text = correct;
	summary.questions_asked.text = numQ;
	var percentage = Math.round(100*correct/numQ);
	summary.perc.text = percentage;
	displayArray="";
	for(i=0;i<wrongUns.length;i++){
		displayArray+=wrongUns[i]+"<br>";
	}
	summary.further_study.htmlText = displayArray;
	elapsed();
	summary.m.text = minutes_elapsed;
	summary.s.text = seconds_elapsed;

	// now the variables to be sent by mail
	summary.realname = me;
	summary.marks = correct;
	summary.asked = numQ;
	summary.timeTakenMin = minutes_elapsed;
	summary.timeTakenSec = seconds_elapsed;
	summary.destination = myMail;// the email address of the person doing the test
	summary.wrongOnes = wrongUnsPhp+"\n";
}
*********************************************************************
Here is the bare bones of the actionscript 2 'end' function:
*********************************************************************
function end() {
	fillForm();
	summary.sendit.onRelease = function() {
		// send variables in form movieclip (the textfields)
		// to email PHP page which will send the mail
		summary.loadVariables("../mail/shapes_sl.php","POST");
		summary.onEnterFrame = function() {
			_root.attachMovie("thanks","thanks",_root.getNextHighestDepth(),{_x:0, _y:0});
			removeMovieClip(summary);
		};
		removeMovieClip("formula");
	};
}
*********************************************************************

The php mailer simply receives the variables and processes them. For example:
**********************************************
$data1=$_POST["realname"];
$data2=$_POST["marks"];
$data3=$_POST["asked"];

***********************************************

All of this worked a dream in Flash, but not using Ruffle. I would love to contribute to the development, but I fear that the programming required is waaay beyond my level. The job you're all doing is fantastic.

If anyone could help with Flash >>> php I would really appreciate it.

Cheers,

Charco

**Comment by n0samu** [comment, marked as answer]
This was discussed more in #2584, and I found the cause of the issue and opened #10545. The problem was fixed soon afterwards so this should be working fine now.
